src/app/modules/shared_modules/keycard_popup/internal/wrong_pin_state.nim

```diff
@@ -131,6 +134,39 @@ method resolveKeycardNextState*(self: WrongPinState, keycardFlowType: string, ke
     if keycardFlowType == ResponseTypeValueKeycardFlowResult:
       controller.setMetadataFromKeycard(keycardEvent.cardMetadata)
       return createState(StateType.PinVerified, self.flowType, nil)
+  if self.flowType == FlowType.ImportFromKeycard:
+    if keycardFlowType == ResponseTypeValueEnterPIN and 
+      keycardEvent.error.len > 0 and
+      keycardEvent.error == ErrorPIN:
+        controller.setRemainingAttempts(keycardEvent.pinRetries)
+        if keycardEvent.pinRetries > 0:
+          return self
+        controller.setKeycardData(updatePredefinedKeycardData(controller.getKeycardData(), PredefinedKeycardData.DisableSeedPhraseForUnlock, add = true))
+        return createState(StateType.MaxPinRetriesReached, self.flowType, nil)
+    if keycardFlowType == ResponseTypeValueEnterPUK and 
+      keycardEvent.error.len == 0:
+        if keycardEvent.pinRetries == 0 and keycardEvent.pukRetries > 0:
+          controller.setKeycardData(updatePredefinedKeycardData(controller.getKeycardData(), PredefinedKeycardData.DisableSeedPhraseForUnlock, add = true))
+          return createState(StateType.MaxPinRetriesReached, self.flowType, nil)
+    if keycardFlowType == ResponseTypeValueKeycardFlowResult:
+      controller.setKeycardUid(keycardEvent.instanceUID)
+      controller.setMetadataForKeycardImport(keycardEvent.cardMetadata)
+      let accountItem = controller.getKeyPairHelper().getAccountsModel().getItemAtIndex(0)
+      if accountItem.isNil:
+        # should never be here (if keycard doesn't contain metadata we should not be able to proceed with this flow)
```

Comment:
same thing, is this an "impossible" condition? we might want to log an error if that's the case